provider/kubeletconfig/kubeletconfig_resource.go

```diff
@@ -0,0 +1,257 @@
+		Attributes: map[string]schema.Attribute{
+			"cluster": schema.StringAttribute{
+				Required:    true,
+				Description: "Identifier of the cluster.",
+				PlanModifiers: []planmodifier.String{
+					common.Immutable(),
+				},
+			},
+			"pod_pids_limit": schema.Int64Attribute{
+				Required: true,
+				Description: "Sets the requested podPidsLimit to be applied as part of the custom " +
+					"KubeletConfig.",
+				Validators: []validator.Int64{
+					PidsLimitValidator{},
+				},
+			},
+		},
+	}
+}
+
+func (k *KubeletConfigResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+
+	plan := k.getKubeletConfigStateFromPlan(ctx, req.Plan, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	clusterId := plan.Cluster.ValueString()
+
+	if err := k.clusterWait.WaitForClusterToBeReady(ctx, clusterId); err != nil {
+		resp.Diagnostics.AddError(
+			"Cluster is not ready",
+			fmt.Sprintf("Cluster with id '%s' is not in the ready plan: %v", clusterId, err),
+		)
+		return
+	}
+
+	if exists, _, err := k.configClient.Exists(ctx, clusterId); err != nil || exists {
+		resp.Diagnostics.AddError(failedToCreateSummary,
+			fmt.Sprintf("KubeletConfig for cluster '%s' may already exist: %v", clusterId, err))
+		return
+	}
+
+	kubeletConfig, err := k.convertStateToApiResource(plan)
+	if err != nil {
+		resp.Diagnostics.AddError(failedToCreateSummary,
+			fmt.Sprintf("Failed to build request to create KubeletConfig on cluster '%s': %v", clusterId, err))
+		return
+	}
+
+	if _, err = k.configClient.Create(ctx, clusterId, kubeletConfig); err != nil {
+		resp.Diagnostics.AddError(failedToCreateSummary,
+			fmt.Sprintf("Failed to create KubeletConfig on cluster '%s': %v", clusterId, err))
+		return
+	}
+
+	k.writeStateToResponse(ctx, plan, &resp.State, &resp.Diagnostics)
+}
+
+func (k *KubeletConfigResource) convertStateToApiResource(state *KubeletConfigState) (*cmv1.KubeletConfig, error) {
+	builder := cmv1.KubeletConfigBuilder{}
+	builder.PodPidsLimit(int(state.PodPidsLimit.ValueInt64()))
+	return builder.Build()
+}
+
+func (k *KubeletConfigResource) convertApiResourceToState(
+	kubeletConfig *cmv1.KubeletConfig, state *KubeletConfigState) {
+	state.PodPidsLimit = types.Int64Value(int64(kubeletConfig.PodPidsLimit()))
+}
+
+func (k *KubeletConfigResource) getKubeletConfigStateFromState(
+	ctx context.Context, state tfsdk.State, diags *diag.Diagnostics) *KubeletConfigState {
+	ks := &KubeletConfigState{}
+	ds := state.Get(ctx, ks)
+	diags.Append(ds...)
+	return ks
+}
+
+func (k *KubeletConfigResource) getKubeletConfigStateFromPlan(
+	ctx context.Context, plan tfsdk.Plan, diags *diag.Diagnostics) *KubeletConfigState {
+	ks := &KubeletConfigState{}
+	ds := plan.Get(ctx, ks)
+	diags.Append(ds...)
+	return ks
+}
+
+func (k *KubeletConfigResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	plan := k.getKubeletConfigStateFromState(ctx, req.State, &resp.Diagnostics)
```

Comment:
Why did you change `state` to `plan` here
The comment wasn't relevant to here
Here you are using the `state` (and not the `plan`)

Reply:
I fixed this and going to push additional commit to this PR, to unblock the merge